src/compiler/checker.ts

```diff
@@ -936,6 +937,25 @@ namespace ts {
             return false;
         }
 
+
+        function checkAndReportErrorForExtendingInterface(errorLocation: Node, name: string): boolean {
+            if (!errorLocation || errorLocation.kind !== SyntaxKind.Identifier ||
+                !errorLocation.parent || !errorLocation.parent.parent ||

```

Comment:
consider this:

```ts
namespace N {
    export interface I { }
}

class C extends N.I {
}
```

you still want to show the error on `I`. the node you get is `HeritageClause` -> `ExpressionWithTypeArguments` -> `PropertyAccessExpression` -> `Identifier`